projects/admin/src/app/guard/can-add-local-fields.guard.ts

```diff
@@ -54,35 +54,24 @@ export class CanAddLocalFieldsGuard implements CanActivate {
   /**
    * Can activate
    * @param next - ActivatedRouteSnapshot
-   * @param state - RouterStateSnapshot
+   * @returns True if authorized access
+   * @throws redirect to error 400
    */
-  canActivate(
-    next: ActivatedRouteSnapshot,
-    state: RouterStateSnapshot): Observable<boolean | UrlTree> | Promise<boolean | UrlTree> | boolean | UrlTree {
-
+  canActivate(next: ActivatedRouteSnapshot): Observable<boolean> {
     const params = next.queryParams;
     if (params.type && params.ref) {
-      /*
-       * Use of User local storage, because the loading of the routing
-       * is done before the load of the application.
-       */
-      const userLocale: IUserLocaleStorage = this._userService.getOnLocaleStorage();
-      if (userLocale) {
-        const organisationPid = userLocale.currentOrganisation;
-        return this._localFieldsApiService.getByResourceTypeAndResourcePidAndOrganisationId(
-          this._translateType(params.type),
-          params.ref,
-          organisationPid
-        ).pipe(map(record => {
-          return record.metadata ? false : true;
-        }));
-      }
-      this._router.navigate(['/errors/401'], { skipLocationChange: true });
+      const type = this._translateType(params.type);
+      return this._localFieldsApiService.getByResourceTypeAndResourcePidAndOrganisationId(
+        type,
+        params.ref,
+        this._userService.user.currentOrganisation
+      ).pipe(map(record => {
+        return record.metadata ? false : true;
```

Comment:
cosmetic : I learned the `!!` operator yesterday to convert anything to boolean : https://brianflove.com/2014-09-02/whats-the-double-exclamation-mark-for-in-javascript/
So you should use  `return !!(record.metadata);` (to be tested)